jib-core/src/main/java/com/google/cloud/tools/jib/async/AsyncDependencies.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.async;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+/**
+ * Builds a list of dependency {@link ListenableFuture}s to wait on before calling a {@link
+ * Callable}.
+ */
+public class AsyncDependencies {
+
+  /**
+   * Initialize with a {@link ListeningExecutorService}.
```

Comment:
See if this can just be `ExecutorService`.

Reply:
The `ListeningExecutorService` is necessary for returning the `ListenableFuture` in `whenAllSucceed`.